spec.md

```diff
@@ -209,19 +209,25 @@ that contains both context and data).
     [RFC 2046](https://tools.ietf.org/html/rfc2046)
 * For Media Type examples see [IANA Media Types](http://www.iana.org/assignments/media-types/media-types.xhtml)
 
-### extensions
-* Type: Map <String, Object>
-* Description: This is for additional metadata and this does not have a
-  mandated structure. This enables a place for custom fields a producer or
-  middleware might want to include and provides a place to test metadata before
-  adding them to the CloudEvents specification. TBD - Determine a shorter
-  prefix for this (e.g. OpenAPI uses “x-”)
+### application-properties
+* Type: <Key Value>
```

Comment:
I think it would be easier to just define this as a Map and say the "value" MUST be a "string" instead of an "Any". That would remove the need for most of the rules specified below because they would be implicit/obvious.  But if you do keep any of the rules then they should use MUST instead of "should".